instagram-app/src/App.js

```diff
@@ -1,14 +1,16 @@
 import React from 'react';
 import PostsPage from './components/PostContainer/PostsPage'
-import withAuth from './components/authentication/withAuthenticate'
+import PrivateRoute from './components/authentication/withAuthenticate'
 import Login from './components/Login/Login'
-
-const ComponentWithAuthenticate=withAuth(PostsPage)(Login)
+import {BrowserRouter as Router,Route, Link, Redirect} from 'react-router-dom'
 
 export default class extends React.Component {
   render(){
     return (
-      <ComponentWithAuthenticate />
+      <Router>
+        <PrivateRoute path='/' component={PostsPage}/>
+        <Route path='/login' component={Login}/>
+      </Router>
```

Comment:
You actually need to use the `exact` flag to your route here. If you notice that both the login page and app are rendering once you get into the actual app (look at the bottom of the page).